### ISSUE TITLE
Impossible to use a new Spyder 4.2+

### ISSUE THREAD

**Issue by andrea83ny** [spyder-ide/spyder#14898]
## Problem Description
Hi, please, I have a Windows machine and Anaconda installation.
My goal is to update spyder from version 4.1.5 to 4.2.3.
All I simply do is I open the Anaconda Prompt, and all I simply do is running "conda install spyde=4.2.3

This triggers the error message: PackageNotFoundError.

So, test number 1 doesn't work.

Now, I manually installed the 4.2.3 installer. Installation was successful. 
However, there's not a single post that shows how to launch this new spyder version.
Just to make things even more clear, using the Windows installer has *not* uninstall the current version of Spyder (4.1.5).
So, this is a total mess.
Is there any way somebody can show clearly and easily how I can use the new Spyder?
I'm a mix between being desperate and about simply to completely uninstall spyder and move to PyCharm.
I hope somebody here could understand my frustration and provide a simple working solution.

### What steps reproduce the problem?

1. 
2. 
3. 

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 4.1.5
* Python version: 
* Qt version: 
* PyQt version: 
* Operating System name/version: 

### Dependencies

```
PASTE DEPENDENCIES HERE
```

**Comment by steff456** [comment]
Hi @andrea83ny,

I understand your frustration, this is the first time we updated the installers and we are trying to see how to enhance the experience of updating them. I guess you installed spyder 4.1.5 in an environment and used the installer to update to 4.2.3. Please remove Spyder from the environment you installed, and use the one of the installer. You should be able to start Spyder normally as a Windows application.

Please let me know if you are able to remove the installation of Spyder 4.1.5 successfully and the version of the installer is working properly :)

**Comment by ccordoba12** [comment]
Hey @andrea83ny, thanks for reporting. You said:

> All I simply do is I open the Anaconda Prompt, and all I simply do is running "conda install spyde=4.2.3

4.2.3 is not available in Anaconda yet. However, you can update to our 4.2.1 version by running the following commands in the Anaconda Prompt:

    conda update anaconda
    conda install spyder=4.2.1

> However, there's not a single post that shows how to launch this new spyder version.

That's right, we'll try to improve our docs about it. However, since this is like a regular Windows application, we create an entry in the Start menu for it. So we expect people try to launch it from it.

> Just to make things even more clear, using the Windows installer has not uninstall the current version of Spyder (4.1.5).
So, this is a total mess.

That's because the application that comes with our installer is separate from Anaconda, that's why it can't uninstall the Anaconda's Spyder.

We understand this is also confusing and we'll try to clarify it in our documentation.

> Is there any way somebody can show clearly and easily how I can use the new Spyder?
I'm a mix between being desperate and about simply to completely uninstall spyder and move to PyCharm.
I hope somebody here could understand my frustration and provide a simple working solution.

Besides uninstalling Spyder (as @steff456 suggested) you also need to connect the Spyder version from our installter to Anaconda, so you can use all the packages that come with it. To do that, please read [these instructions](http://docs.spyder-ide.org/current/faq.html#using-existing-environment).

**Comment by andrea83ny** [comment]
Hi @steff456  @ccordoba12 , thanks for getting back to me, truly appreciated.
Just to make sure I do exactly what you and your team thinks is best, how shall I uninstall spyder?

Just to be 200% clear, if I open Anaconda Navigator right now, the Spyder installation displayed there is still 4.1.5.
(By the way, I tried to update Spyder to 4.2.1 from this anaconda interface, but after of hours loading, nothing happens).

So, my first question is: 
what's the best way to uninstall the spyder version 4.1.5 linked to my Anaconda Navigator?

My second question is:
@ccordoba12 is right, I do see the new Windows installation in the Start Menu.
Is there a way now to "link" this new installation to my Anaconda Navigator?

So basically, can I tell Anaconda to switch from the old installation to the new one?

Thanks,

Andrea

**Comment by andrea83ny** [comment]
I think I have a bigger problem now.
I have Anaconda (and all my packages) installed under C:\bin\anaconda_2021.
I'm not using a virtual environment because of the infinite problems that Spyder was triggering when using a virtual env.
Now, since Spyder 4.2.3 is not "the default" one, it means it doesn't have all the libraries (and their version) available if I use Spyder 4.1.5 installation.

If I go to my Spyder 4.2.3 editor -> Tools -> Preferences -> what Python Interpreter shall I use to make sure it will have the same libraries available with Spyder 4.1.5?
Thanks

**Comment by ccordoba12** [comment]
> Just to make sure I do exactly what you and your team thinks is best, how shall I uninstall spyder?

You need to open the Anaconda Prompt and write there

    conda remove spyder

However, that's only necessary so you don't see multiple Spyder installations around because it doesn't interfere with the Spyder version that comes with our installer.

So if that command takes a lot of time to finish, please close the Anaconda Prompt to stop it.

> Is there a way now to "link" this new installation to my Anaconda Navigator?

No, there's not because the new installation and Anaconda Navigator are completely unrelated.

> So basically, can I tell Anaconda to switch from the old installation to the new one?

No, you can't.

> If I go to my Spyder 4.2.3 editor -> Tools -> Preferences -> what Python Interpreter shall I use to make sure it will have the same libraries available with Spyder 4.1.5?

Please post an screenshot of the available options so we can let you know which one you need to select.

**Comment by Asmedeus998** [comment]
@andrea83ny My solution to fix Spyder 4.2.1 by install it in new enviroment.
run this in anaconda command:
`conda install -c conda-forge spyder`

Make sure your new enviroment doesn't have any previous anaconda.

**Comment by andrea83ny** [comment]
@Asmedeus998 what does it mean "Make sure your new enviroment doesn't have any previous anaconda."?
I have one anaconda installed on my Windows machine. That's all.

**Comment by Asmedeus998** [comment]
> @Asmedeus998 what does it mean "Make sure your new enviroment doesn't have any previous anaconda."?
> I have one anaconda installed on my Windows machine. That's all.

Opps, what I was trying to say before is to create new enviroment and install new spyder using command :
`conda install -c conda-forge spyder`

**Comment by ccordoba12** [comment]
@andrea83ny, you're almost done to have Spyder 4.2.3 working in your machine. Just post the screenshot I mentioned above so we can guide you in the last step. It needs to be like this:

![imagen](https://user-images.githubusercontent.com/365293/110661055-2e7bb680-8192-11eb-8ccd-864b2cb378ba.png)

**Comment by andrea83ny** [comment]
@ccordoba12 I don't know what happened, but the editor doesn't open anymore from the Start Menu. Is there any other way to manually launch it?

**Comment by ccordoba12** [comment]
No, there isn't, sorry. Please try to uninstall it and reinstall it again.

**Comment by andrea83ny** [comment]
@ccordoba12 I did it. Same thing. Spyder doesn't even open.
I checked what the link it's supposed to do, this is what it shows:
C:\bin\Spyder4.2.3\Python\pythonw.exe "C:\bin\Spyder4.2.3\Spyder.launch.pyw"
and both files seem to be there.

**Comment by ccordoba12** [comment]
Closing because this probably no longer applies to our latest version.
